ui/src/app/kvm/views/KVMList/RAMColumn/RAMPopover/RAMPopover.tsx

```diff
@@ -1,33 +1,31 @@
-import PropTypes from "prop-types";
+import type { ReactNode } from "react";
 
 import Popover from "app/base/components/Popover";
-import { formatBytes } from "app/utils";
+import { memoryWithUnit } from "app/kvm/utils";
+import type { Pod, PodMemoryResource } from "app/store/pod/types";
+import { resourceWithOverCommit } from "app/store/pod/utils";
 
 type Props = {
-  allocated: number;
-  children: JSX.Element | string;
-  physical: number;
-  overcommit: number;
+  children: ReactNode;
+  memory: PodMemoryResource;
+  overCommit: Pod["memory_over_commit_ratio"];
 };
 
-const RAMPopover = ({
-  allocated,
-  children,
-  physical,
-  overcommit,
-}: Props): JSX.Element => {
-  const totalMemory = formatBytes(physical * overcommit, "MiB", {
-    binary: true,
-  });
-  const physicalMemory = formatBytes(physical, "MiB", {
-    binary: true,
-  });
-  const allocatedMemory = formatBytes(allocated, "MiB", {
-    binary: true,
-  });
-  const freeMemory = formatBytes(physical * overcommit - allocated, "MiB", {
-    binary: true,
-  });
+const RAMPopover = ({ children, memory, overCommit }: Props): JSX.Element => {
```

Comment:
Same thing with the CPU components, can we dedupe any of the calculations to prevent errors in the future?